variables.tf

```
@@ -9,3 +13,257 @@ variable "tags" {
   description = "User-Defined tags"
   type        = map(string)
 }
+
+########################################
+# Bitbucket Vars
+########################################
+
+variable "admin_email" {
+  description = "email address for administrator"
+  type        = string
+}
+
+variable "admin_password" {
+  description = "initial password to use for administrator"
+  type        = string
+}
+
+variable "data_volume_size" {
+  description = "Size in GB of the data volume"
+  type        = number
+}
+
+variable "dns_prefix" {
+  default     = null
+  description = "the hostname that will be used for bitbucket. This will be combined with the domain in `zone_id` or the value of `domain_name` to form the base url."
+  type        = string
+}
+
+variable "domain_name" {
+  default     = null
+  description = "domain name, which is only used if `zone_id` is not specified to compute the base url"
+  type        = string
+}
+
+variable "license_key" {
+  default     = ""
+  description = "Bitbucket license key (optional, must be a single line)"
+  type        = string
+}
+
+variable "site_name" {
+  default     = "Bitbucket"
+  description = "Bitbucket site name"
+  type        = string
+}
+
+variable "volume_key" {
+  default     = "bitbucket-volume"
+  description = "This value is set to a key on the EBS volume and must be present for the bitbucket instance to be permitted to attach it."
+  type        = string
+}
+
+########################################
+# ASG Vars
+########################################
+
+variable "ami_id" {
+  description = "AMI to build on (must have `ansible-role-atlassian-bitbucket` module installed)"
+  type        = string
+}
+
+variable "asg_additional_iam_policies" {
+  default     = []
+  description = "Additional IAM policies to attach to the  ASG instance profile"
+  type        = list(string)
+}
+
+variable "asg_additional_security_groups" {
+  default     = []
+  description = "Additional security group IDs to attach to ASG instances"
+  type        = list(string)
+}
+
+variable "asg_additional_user_data" {
+  default     = ""
+  description = "Additional User Data to attach to the launch template"
+  type        = string
+}
+
+variable "asg_allow_outbound_egress" {
+  default     = true
+  description = "whether or not the default SG should allow outbound egress"
+  type        = bool
+}
+
+variable "asg_desired_capacity" {
+  default     = 1
+  description = "The number of Amazon EC2 instances that should be running in the group."
+  type        = number
+}
+
+variable "asg_instance_type" {
+  default     = "t3a.micro"
+  description = "Instance type for scim app"
+  type        = string
+}
+
+variable "asg_key_name" {
+  default     = null
+  description = "Optional keypair to associate with instances"
+  type        = string
+}
+
+variable "asg_max_size" {
+  default     = 2
```

```suggestion
  default     = 1
```

The default should be 1. A second instance _can't_ become healthy because it can't grab the EBS volume, so even during deployments the count shouldn't pass 1.